lnwallet/commitment.go

```diff
@@ -149,6 +149,36 @@ func DeriveCommitmentKeys(commitPoint *btcec.PublicKey,
 	return keyRing
 }
 
+// ScriptInfo holds a redeem script and hash.
+type ScriptInfo struct {
+	// PkScript is the outputs' PkScript.
+	PkScript []byte
+
+	// WitnessScript is the full script required to properly redeem the
+	// output with PkSript. This field will only be populated if a PkScript
+	// is p2wsh or p2sh.
+	WitnessScript []byte
+}
+
+// CommitScriptToRemote creates the script that will pay to the non-owner of
+// the commitment transaction, adding a delay to the script based on the
+// channel type.
+func CommitScriptToRemote(_ channeldb.ChannelType, csvTimeout uint32,
+	key *btcec.PublicKey) (*ScriptInfo, error) {
+
+	p2wkh, err := input.CommitScriptUnencumbered(key)
+	if err != nil {
+		return nil, err
+	}
+
+	// Since this is a regular P2WKH, the WitnessScipt and PkScript should
+	// be both seth to the script hash.
+	return &ScriptInfo{
+		WitnessScript: p2wkh,
```

Comment:
the godoc on `WitnessScript` says this will only be populated for `p2wsh` or `p2sh`, does the comment need updating?

Reply:
Fixed: https://github.com/lightningnetwork/lnd/pull/3829/commits/95cac3f2e74a18bdeb664f2bb043a0b21e2bd266